src/rez/utils/pip.py

```diff
@@ -0,0 +1,705 @@
+        `VersionRange`: Equivalent rez version range.
+    """
+    ranges = []
+    parts = specifier.split(',')
+
+    # PEP440 allows spaces, rez doesn't
+    parts = [''.join(x.split()) for x in parts]
+
+    def is_release(rez_ver):
+        parts = rez_ver.split('.')
+        try:
+            _ = int(parts[-1])  # noqa
+            return True
+        except:
+            return False
+
+    # 1 --> 2; 1.2 --> 1.3; 1.a2 -> 1.0
+    def next_ver(rez_ver):
+        parts = rez_ver.split('.')
+        if is_release(rez_ver):
+            parts = parts[:-1] + [str(int(parts[-1]) + 1)]
+        else:
+            parts = parts[:-1] + ["0"]
+        return '.'.join(parts)
+
+    # 1 --> 1.1; 1.2 --> 1.2.1; 1.a2 --> 1.0
+    def adjacent_ver(rez_ver):
+        if is_release(rez_ver):
+            return rez_ver + ".1"
+        else:
+            parts = rez_ver.split('.')
+            parts = parts[:-1] + ["0"]
+            return '.'.join(parts)
+
+    def convert_spec(spec):
+        def parsed_rez_ver():
+            v = spec
+            for substr in ('!', '=', '~', '<', '>', ".*"):
+                v = v.replace(substr, '')
+            return pip_to_rez_version(v)
+
+        def fmt(txt):
+            v = parsed_rez_ver()
+            vnext = next_ver(v)
+            vadj = adjacent_ver(v)
+            return txt.format(V=v, VNEXT=vnext, VADJ=vadj)
+
+        # ==1.* --> 1
+        if spec.startswith("==") and spec.endswith(".*"):
+            return fmt("{V}")
+
+        # ==1 --> 1+<1.1
+        if spec.startswith("=="):
+            return fmt("{V}+<{VADJ}")
+
+        # >=1 --> 1+
+        if spec.startswith(">="):
+            return fmt("{V}+")
+
+        # >1 --> 1.1+
+        if spec.startswith(">"):
+            return fmt("{VADJ}+")
+
+        # <= 1 --> <1.1
+        if spec.startswith("<="):
+            return fmt("<{VADJ}")
+
+        # <1 --> <1
+        if spec.startswith("<"):
+            return fmt("<{V}")
+
+        # ~=1.2 --> 1.2+<2; ~=1.2.3 --> 1.2.3+<1.3
+        if spec.startswith("~="):
+            v = Version(parsed_rez_ver())
+            v = v.trim(len(v) - 1)
+            v_next = next_ver(str(v))
+            return fmt("{V}+<" + v_next)
+
+        # !=1.* --> <1|2+; !=1.2.* --> <1.2|1.3+
+        if spec.startswith("!=") and spec.endswith(".*"):
+            return fmt("<{V}|{VNEXT}+")
+
+        # !=1 --> <1|1.1+; !=1.2 --> <1.2|1.2.1+
+        if spec.startswith("!="):
+            return fmt("<{V}|{VADJ}+")
+
+        raise PackageRequestError(
+            "Don't know how to convert PEP440 specifier %r into rez equivalent",
+            specifier
+        )
+
+    for part in parts:
+        ranges.append(convert_spec(part))
+
+    # AND together ranges
+    total_range = VersionRange(ranges[0])
+
+    for range_ in ranges[1:]:
+        range_ = VersionRange(range_)
+        total_range = total_range.intersection(range_)
+
+        if total_range is None:
+            raise PackageRequestError(
```

Comment:
@nerdvegas I'm having difficulties to find when this happens. Could you enlighten me with an example please?